core/inputs/cascading-config.ts

```diff
@@ -0,0 +1,108 @@
+'use strict'
+
+import { type AlbAlarmProperties } from '../alarms/alb'
+import { type AlbTargetAlarmProperties } from '../alarms/alb-target-group'
+import { type ApiGwAlarmProperties } from '../alarms/api-gateway'
+import { type AppSyncAlarmProperties } from '../alarms/appsync'
+import { type DynamoDbAlarmProperties } from '../alarms/dynamodb'
+import { type EcsAlarmsConfig } from '../alarms/ecs'
+import { type EventsAlarmsConfig } from '../alarms/eventbridge'
+import { type KinesisAlarmProperties } from '../alarms/kinesis'
+import { type SnsAlarmsConfig } from '../alarms/sns'
+import { type SqsAlarmsConfig } from '../alarms/sqs'
+import { type SfAlarmsConfig } from '../alarms/step-functions'
+import { type AllAlarmsConfig } from '../alarms/default-config-alarms'
+import {
+  type DashboardConfig, type DashboardBodyProperties, type LambdaDashboardBodyProperties, type ApiGwDashboardBodyProperties, type SfDashboardBodyProperties, type DynamoDbDashboardBodyProperties,
+  type KinesisDashboardBodyProperties, type SqsDashboardBodyProperties, type EcsDashboardBodyProperties, type SnsDashboardBodyProperties, type RuleDashboardBodyProperties,
+  type AlbDashboardBodyProperties, type AlbTargetDashboardBodyProperties, type AppSyncDashboardBodyProperties
+} from '../dashboards/default-config-dashboard'
+import { type LambdaFunctionAlarmPropertiess } from '../alarms/lambda'
+import { type Statistic } from '../cf-template'
+import { type AlarmProperties } from 'cloudform-types/types/cloudWatch/alarm'
+
+const MAX_DEPTH = 10
+
+type ConfigNode = DashboardConfig | AllAlarmsConfig
+
+interface ParentNode {
+  DashboardBodyProperties?: DashboardBodyProperties
+  AlarmProperties?: AlarmProperties
+}
+
+interface TimeRange {
+  start: string
+  end: string
+}
+
+export interface Widgets {
+  ActionsEnabled?: boolean // remove later ? mark
+  Statistic?: Statistic[]
+  Lambda?: LambdaDashboardBodyProperties
+  ApiGateway?: ApiGwDashboardBodyProperties
+  States?: SfDashboardBodyProperties
+  DynamoDB?: DynamoDbDashboardBodyProperties
+  Kinesis?: KinesisDashboardBodyProperties
+  SQS?: SqsDashboardBodyProperties
+  ECS?: EcsDashboardBodyProperties
+  SNS?: SnsDashboardBodyProperties
+  Events?: RuleDashboardBodyProperties
+  ApplicationELB?: AlbDashboardBodyProperties
+  ApplicationELBTarget?: AlbTargetDashboardBodyProperties
+  AppSync?: AppSyncDashboardBodyProperties
+}
+
+export interface DashboardsCascade {
```

Comment:
We are mixing two things in this file now:
1. Types for SLIC Watch configuration
2. The `cascade` function

The two can be separate since the action of `cascading` is independent from the types.
We can also remove the word `Cascade` from these type names, since the configuration is independent of the process of cascading

`DashboardsCascade` -> `DashboardConfig` (or similar)
`AlarmsCascade` -> `AlarmConfig` (or similar)


The types `AllAlarmsConfig` and `AlarmsCascade` seem to be pretty much the same thing, so can we merge them into one? 

Maybe for clarity, the type names should be `SlicWatchAlarmsConfig` and `SlicWatchDashboardConfig`? This would differentiate them from CloudFormation/AWS alarm and dashboard configuration syntax.